modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidatorTestCase.java

```diff
@@ -281,6 +300,46 @@ public void connectionProviderWithRepeatedParameterName() {
   }
 
   @Test
+  public void sourceWithRepeatedParameterNameWithinCallback() {
+    exception.expect(IllegalModelDefinitionException.class);
+    ParameterModel offending = getParameter(SIMPLE_PARAM_NAME, String.class);
+    SourceCallbackModel sourceCallbackModel = mock(SourceCallbackModel.class);
+    when(sourceCallbackModel.getAllParameterModels()).thenReturn(asList(simpleConnectionProviderParam, offending));
+    when(sourceModel.getErrorCallback()).thenReturn(Optional.of(sourceCallbackModel));
+    validate();
+  }
+
+  @Test
+  public void sourceWithRepeatedParameterNameAmongCallbackAndSource() {
+    exception.expect(IllegalModelDefinitionException.class);
+    ParameterModel offending = getParameter(SIMPLE_PARAM_NAME, String.class);
+    SourceCallbackModel sourceCallbackModel = mock(SourceCallbackModel.class);
+    when(sourceCallbackModel.getAllParameterModels()).thenReturn(asList(simpleConfigParam));
+
+    ParameterGroupModel group = mock(ParameterGroupModel.class);
+    when(group.getName()).thenReturn(DEFAULT_GROUP_NAME);
+    when(group.getModelProperty(ParameterGroupModelProperty.class)).thenReturn(empty());
+    when(group.getParameterModels()).thenReturn(asList(offending));
+
+    SourceModel sourceModel = new ImmutableSourceModel(SOURCE_NAME, "", false, asList(group), null, null,
```

Comment:
This had to done in order to test SourceModel interface default method. Couldn't be done with mockito.